### ISSUE TITLE
[Question] List of features?

### ISSUE THREAD

**Issue by Kikobeats** [decaffeinate/decaffeinate#348]
This projects looks great, but without a list of supported features I can't considered it a real alternative.

something like http://node.green/ based in decaffeinate semver 😄

**Comment by alangpierce** [comment]
Agreed that that kind of thing would be really useful! However, my impression is that a "list of supported features" isn't quite the right way to understand the state of the project. Most of the problems I run into now are due to bugs when converting complicated real-world code, rather than specific language features not being implemented. A similar "state of Decaffeinate dashboard" idea I've had is to show how well Decaffeinate does at converting major open source CoffeeScript projects.

Since Decaffeinate is designed for a supervised one-time conversion, I think the problem is a little different from most transpilers/interpreters. Sometimes it's better to focus on generating nicer JS for common features than it is to handle a case that only appears once per 100,000 lines. Ideally, Decaffeinate would be able to handle everything, but when it can't handle some obscure feature, it doesn't mean you can't use Decaffeinate, it just means that you need to spend some time up-front to modify your CoffeeScript code to be Decaffeinate-friendly.

To give a sense of the state of the project, the codebase I'm wanting to convert has about 1300 files and about 200,000 lines of code, and Decaffeinate runs successfully  on all but about 200 files. Of those 200 files, about 75% of them fail due to crashes that are hard to classify (e.g. parens and curly braces get added out of order, so the generated JS doesn't parse), and about 25% fail due to cases that are known to not be supported yet (the most common of which is `for` expressions with a body that can't be treated as an expression). For those 200 files, I expect it would take maybe a few minutes per file to use the [repl](http://decaffeinate-project.org/repl/) to tweak them to be Decaffeinate-friendly. Once all of that is done, I expect that I would need to spent a significant amount of time fixing style/lint issues in the resulting JavaScript code and verifying the correctness of the resulting code.

**Comment by Eden-Sun** [comment]
@alangpierce I also thought Decaffeinate is designed to be an better transpiler and save coffee dying.

**Comment by alangpierce** [comment]
@Eden-Sun I didn't think that was a goal, but @eventualbuddha is the author of the project and has the real vision. Certainly, if decaffeinate becomes mature enough, it could be used as an alternative to the CoffeeScript compiler, but I think decaffeinate is generally a little less focused on 100% correctness and more focused on preserving comments and formatting and making the output look nice. Also, my understanding is that decaffeinate is focused on transforming CoffeeScript as it exists today, and it would be out of scope to add any improvements to the CoffeeScript language.

There's a lengthy discussion of the idea of a CoffeeScript-to-ES6 transpiler (using ES6 features to improve CoffeeScript) here:
https://github.com/jashkenas/coffeescript/issues/4078

And it looks like "CoffeeScript 6" is a recent effort more directly focused on that kind of thing:
https://github.com/coffeescript6/discuss

**Comment by eventualbuddha** [comment]
Thanks for your replies, @alangpierce. You represented my position on the matter accurately.

I do not intend to have decaffeinate be used as an alternative to the official CoffeeScript compiler (a la CoffeeScriptRedux). Instead, its purpose is to aid those who wish to migrate away from CoffeeScript and onto JavaScript with modern features (ES6 & 7). Therefore, as @alangpierce said, it focuses less on 100% correctness and more on creating readable code you might have written by hand if you'd done it in JavaScript to begin with.

> I also thought Decaffeinate is designed to be an better transpiler and save coffee dying

I'm sorry if I failed to convey my intentions appropriately. I'm working on a new homepage for the project and will try to make the point clearer. I emphatically *do not* want to save CoffeeScript from dying. In fact, I'd like to hasten its death. I totally approve of language experiments and, as I said, I believe CoffeeScript helped move the JavaScript language forward. However, it's time to move on.

Anyone who wants to stay on CoffeeScript and support that platform should do so by participating in the discussions linked in @alangpierce's last comment. I personally think the ship has sailed, however.

**Comment by eventualbuddha** [comment]
I've updated http://decaffeinate-project.org/ to better explain the project, hopefully.

**Comment by Kikobeats** [comment]
👍
